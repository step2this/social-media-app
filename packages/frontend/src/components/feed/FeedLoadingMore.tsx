import React from 'react';

export interface FeedLoadingMoreProps {
  loading: boolean;
}

export const FeedLoadingMore: React.FC<FeedLoadingMoreProps> = ({ loading }) => {
  if (!loading) return null;
  return (
    <div className="feed-loading-more" aria-live="polite">
      <div className="spinner"></div>
      <p>Loading more...</p>
    </div>
  );
};
