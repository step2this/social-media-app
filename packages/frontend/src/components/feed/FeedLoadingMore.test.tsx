import { describe, it, expect } from 'vitest';
import { render, screen } from '@testing-library/react';
import { FeedLoadingMore } from './FeedLoadingMore';

describe('FeedLoadingMore', () => {
  it('shows spinner when loading=true', () => {
    const { container } = render(<FeedLoadingMore loading={true} />);
    const spinner = container.querySelector('.spinner');
    expect(spinner).toBeInTheDocument();
  });

  it('hides spinner when loading=false', () => {
    const { container } = render(<FeedLoadingMore loading={false} />);
    expect(container.firstChild).toBeNull();
  });

  it('displays "Loading more..." text when loading', () => {
    render(<FeedLoadingMore loading={true} />);
    expect(screen.getByText('Loading more...')).toBeInTheDocument();
  });

  it('has correct CSS classes when loading', () => {
    const { container } = render(<FeedLoadingMore loading={true} />);
    const loadingDiv = container.querySelector('.feed-loading-more');
    expect(loadingDiv).toBeInTheDocument();
    expect(loadingDiv).toHaveClass('feed-loading-more');
  });

  it('matches snapshot (loading state)', () => {
    const { container } = render(<FeedLoadingMore loading={true} />);
    expect(container.firstChild).toMatchSnapshot();
  });

  it('matches snapshot (not loading state)', () => {
    const { container } = render(<FeedLoadingMore loading={false} />);
    expect(container.firstChild).toMatchSnapshot();
  });

  it('has accessible aria-live region when loading', () => {
    const { container } = render(<FeedLoadingMore loading={true} />);
    const loadingDiv = container.querySelector('.feed-loading-more');
    expect(loadingDiv).toHaveAttribute('aria-live', 'polite');
  });

  it('renders without crashing', () => {
    const { container } = render(<FeedLoadingMore loading={false} />);
    expect(container).toBeInTheDocument();
  });
});
