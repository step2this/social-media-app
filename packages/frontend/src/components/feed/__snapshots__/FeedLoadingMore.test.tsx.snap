// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`FeedLoadingMore > matches snapshot (loading state) 1`] = `
<div
  aria-live="polite"
  class="feed-loading-more"
>
  <div
    class="spinner"
  />
  <p>
    Loading more...
  </p>
</div>
`;

exports[`FeedLoadingMore > matches snapshot (not loading state) 1`] = `null`;
